pkgs/development/python-modules/torchrl/default.nix

```diff
@@ -0,0 +1,116 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, ninja
+, setuptools
+, wheel
+, which
+, cloudpickle
+, numpy
+, torch
+, ale-py
+, gymnasium
+, pygame
+, torchsnapshot
+, mujoco
+, moviepy
+, git
+, hydra-core
+, tensorboard
+, tqdm
+, wandb
+, packaging
+, tensordict
+, flaky
+, pytestCheckHook
+, pyyaml
+, scipy
+}:
+
+buildPythonPackage rec {
+  pname = "torchrl";
+  version = "0.2.1";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "pytorch";
+    repo = "rl";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-Y3WbSMGXS6fb4RyXk2SAKHT6RencGTZXM3tc65AQx74=";
+  };
+
+  nativeBuildInputs = [
+    ninja
+    setuptools
+    wheel
+    which
+  ];
+
+  propagatedBuildInputs = [
+    cloudpickle
+    numpy
+    packaging
+    tensordict
+    torch
+  ];
+
+  passthru.optional-dependencies = {
+    atari = [
+      ale-py
+      gymnasium
+      pygame
+    ];
+    checkpointing = [
+      torchsnapshot
+    ];
+    gym-continuous = [
+      gymnasium
+      mujoco
+    ];
+    rendering = [
+      moviepy
+    ];
+    utils = [
+      git
+      hydra-core
+      tensorboard
+      tqdm
+      wandb
+    ];
+  };
+
+  preBuild = ''
+    # torchrl needs to create a folder to store datasets
+    export D4RL_DATASET_DIR=$(mktemp -d)
+  '';
+
+  pythonImportsCheck = [
+    "torchrl"
+    "torchrl._torchrl"
+  ];
+
+  preCheck = ''
+    # We have to delete the source because otherwise it is used intead the installed package.
+    rm -rf torchrl
+  '';
+
+  nativeCheckInputs = [
+    flaky
+    gymnasium
+    pytestCheckHook
+    pyyaml
+    scipy
+  ]
+  ++ passthru.optional-dependencies.checkpointing
+  ;
+
+  meta = with lib; {
+    description = "A modular, primitive-first, python-first PyTorch library for Reinforcement Learning";
+    homepage = "https://github.com/pytorch/rl";
+    license = licenses.mit;
```

Comment:
Is it a dual license for MIT and BSD?
https://github.com/pytorch/rl/blob/1bb192e0f3ad9e7b8c6fa769bfa3bb9d82ca4f29/setup.py#L230
https://github.com/pytorch/rl/blob/1bb192e0f3ad9e7b8c6fa769bfa3bb9d82ca4f29/setup.py#L257C42-L257C42

Reply:
Indeed, the pypi page confirms this.
I added both licenses.

Reply:
Oh, I just saw your other comment: switching back to MIT only.